documentation/puppet_server_metrics_performance.markdown

```diff
@@ -38,13 +38,17 @@ If wait time increases but borrow time stays the same, your Server infrastructur
 
 #### Adding more JRubies
 
-If you need to add JRubies, remember that Puppet Server is tuned by default to use one fewer than your total number of CPUs, with a maximum of 4 CPUs, for the number of available JRubies. You can change this by setting `max-active-instances` in [`puppetserver.conf`][], under the `jruby-puppet` section. Don' set this value to something greater than the number of CPUs, because if there aren't enough processes available, it won't matter how many JRubies in the pool there are — the server won't be able to use them.
+If you must add JRubies, remember that Puppet Server is tuned by default to use one fewer than your total number of CPUs, with a maximum of 4 CPUs, for the number of available JRubies. You can change this by setting `max-active-instances` in [`puppetserver.conf`][puppetserver.conf], under the `jruby-puppet` section.
 
-Each JRuby also has a certain amount of memory overhead that it needs in order to be able to load both Puppet's Ruby code and your Puppet code. In other words, your available memory limits how much Puppet code you can process. As a general rule, adding a JRuby requires a bare minimum of 40MB of memory --- the amount of memory for the scripting container, plus Puppet's Ruby code, plus an additional 5MB) --- just to compile an almost-empty catalog. For real-world catalogs, add at least an additional 15MB per additional JRuby.
+Each JRuby also has a certain amount of persistent memory overhead required in order to load both Puppet's Ruby code and your Puppet code. In other words, your available memory sets a baseline limit to how much Puppet code you can process. Catalog compilation can consume more memory, and Puppet Server's total memory usage depends on the number of agents being served, how frequently those agents check in, how many resources are being managed on each agent, and the complexity of the manifests and modules in use.
+
+As a general rule, adding a JRuby requires a bare minimum of 40MB of memory under JRuby 1.7, and at least 60MB under JRuby9k if the `jruby-puppet.compile-mode` setting in [`puppetserver.conf`][puppetserver.conf] is set to `off` --- the amount of memory for the scripting container, plus Puppet's Ruby code, plus additional memory overhead --- just to compile an nearly empty catalog.
+
+For real-world catalogs, you can generally add at least an additional 15MB for each JRuby you add. We calculated this amount by comparing a minimal catalog compilation to a catalog with a [basic role](https://github.com/puppetlabs/puppetlabs-puppetserver_perf_control/blob/production/site/role/manifests/by_size/small.pp) that installs Tomcat and Postgres servers. You can calculate a similar value unique to your infrastructure by measuring `puppetserver` memory usage during your infrastructure's catalog compilations and comparing it to compiling a minimal catalog for a similar number of nodes.
```

Comment:
I believe the "512 MB" number was derived more from empirical experience of the memory requirements of a system under load whereas the numbers here pertain to minimum persistent memory usage (and tend to discount active / peak request load).  I don't think any of this new information would displace the historical "512 MB" recommendation.

@Sharpie - do you think mentioning this level of "minimum persistent usage" info is useful to external users or would you suggest that we jettison it completely?  I definitely agree with you that it would be good to avoid giving users information that they may see as conflicting.